### DISCUSSION TITLE
Spatial Multilevel Models?

### DISCUSSION THREAD

**Discussion by rafavdz** [ConnorDonegan/geostan#20] [General]
Hi Conor, 

Many thanks for this great contribution. I was wondering if it is possible to fit 'multilevel' models, where individual observations are within spatial areal units. 

Something like this:

![image](https://github.com/ConnorDonegan/geostan/assets/69197515/a5e1a40c-fe76-4325-9489-5f9692a7c04a)

This model is referred in the CARBayes package documentation as S.CARmultilevel(). 

Thanks,
Rafa

**Comment by rafavdz** [comment]
Sorry for the incomplete post. Further information about these types of models here: http://cran.nexr.com/web/packages/CARBayes/vignettes/CARBayes.pdf

**Reply by ConnorDonegan** [reply]
Hi Rafa,

Thanks for the kind words. It is not currently implemented in geostan but I can definitely consider adding something like that. I'll have to look into it a bit - its possible that it can be implemented without upsetting much of the code base, in which case I could create a new branch for this or add it to a new release sometime over the coming months. Do you have a particular use in mind? It might help to know.

**Reply by rafavdz** [reply]
Hi @ConnorDonegan 

This would be fantastic! really looking forward to seeing such an update. I've been working with these types of models for some time to study/model property prices, where houses are within neighbourhoods which are spatially autocorrelated. You can see an example in this paper <http://dx.doi.org/10.1016/j.spasta.2017.01.002>. Also, I have used a few of them for my [PhD thesis](https://theses.gla.ac.uk/83588/1/2023verduzco-torresphd.pdf) in Section 5.4.3 and 7.3.4. I've used INLA to fit these models (with some limitations), but I'd be really keen to have more alternatives. 

Thanks a lot! 

Rafa

**Reply by ConnorDonegan** [reply]
Alright, thanks for the references. 

I may have an update later this week on whether I can put this into a new branch or not and if so how long it may take.

I've had a somewhat similar experience as reported in the paper by Bivand and others. I've been working on this kind of model for survival analysis and found that the ICAR model works well (i.e. samples well) but adding on additional terms at the district level (an autocorrelation parameter for the proper CAR model or SAR model) causes the sampling to basically fail. In the survival literature, I haven't seen anyone create software that uses a more 'proper' or complete spatial model (CAR or SAR or BYM) for the district-level - only the ICAR trend is added on.

**Reply by ConnorDonegan** [reply]
@rafavdz, after looking this over, I don't think it makes sense to add it to a geostan branch but I think that I may be able to extend the project I'm already working on, for accelerated failure time models, for this purpose. 

Apart from some plotting methods and censoring (which can be ignored), the AFT models are just GLMs which I've added a shared spatial trend to. So I should be able to extend it by adding options for different spatial model types. I'm trying to have it on a public github repo within two to three months. I'll let you know when there's any news about this (hopefully only positive news). 

Thanks for the request, I think this will make it a more interesting project

**Reply by rafavdz** [reply]
It makes sense. Thanks for the time and effort looking at this, @ConnorDonegan! I'm really looking forward to seeing this implementation. Current options in R are quite limited. So, this will make a great contribution. 

Thanks again!
